site/src/components/AuthProvider/AuthProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import { useActor, useInterpret } from "@xstate/react"
+import { createContext, FC, PropsWithChildren, useContext } from "react"
+import { authMachine } from "xServices/auth/authXService"
+import { ActorRefFrom } from "xstate"
+
+interface AuthProviderContextValue {
+  authService: ActorRefFrom<typeof authMachine>
+}
+
+const AuthProviderContext = createContext<AuthProviderContextValue | undefined>(
```

Comment:
NIT: I would probably just name this `AuthContext`

Reply:
But there is a difference. The AuthContext is the one created from React.createContext, and the Provider is the wrapper around it that define the values like data and additional functions

Reply:
I get the difference - I would name it `AuthContext` for brevity, and also because it's named after the thing it's created by: a hook called `createContext`. You see the React docs using this pattern [here](https://beta.reactjs.org/reference/react/useContext). The interface you defined describes the values available on the context (you're right tho - they are enabled through a provider). That being said, if you prefer the longer name, I'm down. We should just establish a pattern since we'll be using React state more frequently.

Reply:
Ahh I see. But in this case the Context is being used directly. We can think like Material UI that has a wrapper around the their theme context and named it ThemeProvider as example